server/src/main/java/org/opensearch/action/support/nodes/BaseNodeResponse.java

```diff
@@ -67,6 +67,6 @@ public DiscoveryNode getNode() {
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
-        node.writeTo(out);
+        node.writeToWithAttribute(out);
```

Comment:
Does it mean we still will have to write the attribute for APIs that don't need the attributes?

Reply:
We are writing the attribute in the final api response to not alter anything in the api response structure.